ideascube/serveradmin/backup.py

```diff
@@ -113,6 +128,11 @@ def delete(self):
 
     @classmethod
     def create(cls, format=None):
+        format = format or Backup.FORMAT
+        if format not in Backup.SUPPORTED_FORMATS_AT_CREATION:
+            raise ValueError(
+                _("Format {} is not supported at creation").format(format)
+            )
         name = make_name(format)
         backup = Backup(name)
```

Comment:
So we create a `Backup` passing it just a `name`, even though we already know its format.

And then, we have a `guess_file_format` method which has a mapping of extensions to format, and from the the `name` deducts the format.

Why not just store the format as an attribute of the `Backup` instance?

I know, that's not something you touched here. :smile: